### DISCUSSION TITLE
Using the Block Bindings editor API (JS) in WP 6.7

### DISCUSSION THREAD

**Discussion by justintadlock** [WordPress/developer-blog-content#316] [Topic Ideas]
WordPress 6.7 exposes a partial set of JS-based editor functions for working with Block Bindings: https://github.com/WordPress/gutenberg/pull/65713

- `registerBlockBindingsSource`: to register a source in the client.
- `unregisterBlockBindingsSource`: to unregister an existing source.
- `getBlockBindingsSource`: to get a specific registered source and its properties.
- `getBlockBindingsSources`: to get all the registered sources in the client.

These are now stable, public functions for theme and plugin authors to extend how Block Bindings work in their projects, especially custom Block Binding sources. I'm proposing a tutorial that walks developers through using these APIs to manipulate bindings in the editor.

There is an additional `getFieldsList` property that will also be public in the Gutenberg plugin only, which could also be a showcase of what's to come in the future.

**Comment by justintadlock** [comment]
This topic has been approved. Ticket: https://github.com/WordPress/developer-blog-content/issues/317

**Comment by bph** [comment]
Published: [Getting and setting Block Binding values in the Editor](https://developer.wordpress.org/news/2024/10/29/getting-and-setting-block-binding-values-in-the-editor/)
